lib/rules/no-object-constructor.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @fileoverview Rule to disallow calls to the `Object` constructor without an argument
+ * @author Francesco Trotta
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const { getVariableByName, isArrowToken } = require("./utils/ast-utils");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('../shared/types').Rule} */
+module.exports = {
+    meta: {
+        type: "suggestion",
+
+        docs: {
+            description: "Disallow calls to the `Object` constructor without an argument",
+            recommended: false,
+            url: "https://eslint.org/docs/latest/rules/no-object-constructor"
+        },
+
+        hasSuggestions: true,
+
+        schema: [],
+
+        messages: {
+            preferLiteral: "The object literal notation {} is preferable.",
+            useLiteral: "Replace with '{{replacement}}'."
+        }
+    },
+
+    create(context) {
+
+        const sourceCode = context.sourceCode;
+
+        /**
+         * Determines whether or not an object literal that replaces a specified node needs to be enclosed in parentheses.
+         * @param {ASTNode} node The node to be replaced.
+         * @returns {boolean} Whether or not parentheses around the object literal are required.
+         */
+        function needsParentheses(node) {
+            let exprStatNode = node;
+
+            while (exprStatNode && exprStatNode.type !== "ExpressionStatement") {
+                exprStatNode = exprStatNode.parent;
+            }
+            if (exprStatNode && node.range[0] === exprStatNode.range[0]) {
+                return true;
+            }
+
+            const prevToken = sourceCode.getTokenBefore(node);
+
+            if (prevToken && isArrowToken(prevToken)) {
+                return true;
+            }
+
+            return false;
+        }
+
+        /**
+         * Reports on nodes where the `Object` constructor is called without arguments.
+         * @param {ASTNode} node The node to evaluate.
+         * @returns {void}
+         */
+        function check(node) {
+            if (node.callee.name !== "Object" || node.arguments.length) {
```

Comment:
```suggestion
            if (node.callee.type !== "Identifier" || node.callee.name !== "Object" || node.arguments.length) {
```

It's always good to check the `type`, since other nodes may have the same property name.